src/pages/Books.js

```diff
@@ -10,28 +11,60 @@ const Books = () => {
   // Get the books array from the store state
   const books = useSelector((state) => state.books.books);
 
+  // Fetch books from Bookstore API when the component has rendered
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, [dispatch]);
+
   // Function to handle adding a new book
-  const handleAddBook = (event) => {
+  const handleAddBook = async (event) => {
     // Prevent the default form submission behavior
     event.preventDefault();
 
     // Get the input values and create an ID for the new book
     const titleInput = document.querySelector('#titleInput');
     const authorInput = document.querySelector('#authorInput');
-    const itemId = `${titleInput.value}-${authorInput.value}`;
+    const itemId = `${titleInput.value}_${authorInput.value}`;
+
+    try {
+      const newBook = {
+        item_id: itemId,
+        title: titleInput.value,
+        author: authorInput.value,
+        category: 'Fiction',
+      };
 
-    // Dispatch an action to add the new book to the store
-    dispatch(addBook({ item_id: itemId, title: titleInput.value, author: authorInput.value }));
+      // Send a POST request to the Bookstore API to add the new book
+      await axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/LHtDguTMI5Nwg9HRI1l3/books', newBook);
 
-    // Clear the input fields
-    titleInput.value = '';
-    authorInput.value = '';
+      // Dispatch an action to add the new book to the store
+      dispatch(addBook(newBook));
+
+      // Clear the input fields
+      titleInput.value = '';
+      authorInput.value = '';
+    } catch (error) {
+      const errorMessageDiv = document.createElement('div');
+      errorMessageDiv.textContent = 'Error';
+      document.body.appendChild(errorMessageDiv);
+    }
   };
 
   // Function to handle removing a book
-  const handleRemoveBook = (itemId) => {
-    // Dispatch an action to remove the book with the given ID from the store
-    dispatch(removeBook(itemId));
+  const handleRemoveBook = async (itemId) => {
+    try {
+      // Send a DELETE request to the Bookstore API to remove the book with the given ID
+      await axios.delete(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/LHtDguTMI5Nwg9HRI1l3/books/${itemId}`, {
+        data: { item_id: itemId },
+      });
+
+      // Dispatch an action to remove the book with the given ID from the store
+      dispatch(removeBook(itemId));
+    } catch (error) {
+      const errorMessageDiv = document.createElement('div');
+      errorMessageDiv.textContent = 'Error';
+      document.body.appendChild(errorMessageDiv);
+    }
```

Comment:
- [ ] The same happens with the remove, as this requires asynchronous code, please consider using a thunk! 🤓 🔧 

```suggestion
  const handleRemoveBook = async (itemId) => {
    
      
      
        
      

      // Dispatch an action to remove the book with the given ID from the store
      dispatch(removeBook(itemId));
    
      
      
      
    
```